### PR TITLE
Add contribution guidelines

### LINKED ISSUE #2: Add developer guidelines document
Required:
- [x] make sure data is somewhere appropriate.
- [x] packaging

Optional
- [x] testing
- [x] continuous integration
- [ ] coverage
- [ ] binder

Guidance
- [x] Licenses
- [x] README sections

### PR DESCRIPTION
**Pull request by jatkinson1000**
Closes #2 

Add a file contributing.md that sets out how to request a new code/repo, the base requirements, and other features.

### CODE REVIEW

File: contributing.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+All codes **must** have a suitable
+[open-source license](https://opensource.org/licenses/) attached to them.  
+If your code does not already have one we suggest the
+[MIT License](https://mit-license.org/).
+However, please check with any relevant institutions or funding bodies as they may
+have restrictions on the licenses that may be used.
+
+If you are unsure about this please reach out to us and we can provide guidance.  
+A introduction to software licenses can be found in a
```
Comment (lcimoli) [inline contributing.md]: An introduction (typo)

File: contributing.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+[Code for Thought podcast episode](https://codeforthought.buzzsprout.com/1326658/11564212)
+and there is advice on selecting at [choosealicense.com](https://choosealicense.com/).
+
+
+### Readme
+
+Your code should come with an appropriate Readme file.  
+This should ideally be written in [Markdown](https://en.wikipedia.org/wiki/Markdown)
+and stored at the top level as `README.md`.
+
```
Comment (lcimoli) [inline contributing.md:93]: Add sentence: "The Readme file should include the following information:"

File: contributing.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+This document is designed to guide those who have a new code that they wish to bring
+to the GO-SHIP Organisation.
+For details on how to contribute to an existing repository please see the documentation
+of that specific repository for guidance.
+
+
+## Pre-requisites
+
+If you have a code that you would like to add to the organisation please
```
Comment (lcimoli) [inline contributing.md:23]: Should we specify "if you have a code in any programming language (e.g. Python, R, MatLab, etc.)" ?

**Review by lcimoli — COMMENTED** [review]
Just a few minor comments, otherwise reads well and I think inlcudes all needed information - thank you!!

File: contributing.md (hunk 1 above)
Comment (jatkinson1000) [reply contributing.md]: Done.

File: contributing.md (hunk 2 above)
Comment (jatkinson1000) [reply contributing.md:93]: Done.

File: contributing.md (hunk 3 above)
Comment (jatkinson1000) [reply contributing.md:23]: Done.

**Comment by jatkinson1000** [comment]
Hi @lcimoli 
I have added a bunch of information to the contribution guidelines.
I tried to provide comprehensive guidance for people to do things without being too restrictive or overwhelming which was tricky.
What do you think?
If you can review we can hopefully merge soon and then get someone to try following the guidelines.

File: contributing.md [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+
+## Desirables
+
+In addition to the above the following are desirable qualities for research software.
+If you would like assistance with adding these to your code, either before or after the
+initial upload, please ask for assistance.
+
+### Packaging
+
+If you can package your code to make it easy for to install this is very useful.
```
Comment (lcimoli) [inline contributing.md]: to make it easy to install (remove for)

File: contributing.md [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+to upload the code to GitHub.
+
+You can then open a [pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request)
+from the repository webpage on GitHub to ask to add your code to the main branch.
+
+If you are unsure about any part of this process we are happy to guide you through it.
+
+### Further developments
+
+Once you have performed the initial upload of the code to the main branch of your
```
Comment (lcimoli) [inline contributing.md]: remove "to the main branch"...? we suggested the initial upload is done on a first-commit branch, right?

**Review by lcimoli — COMMENTED** [review]
Hi Jack, this looks great - thank you so much for putting it together!
I just made a couple of minor comments. Otherwise, I think it is good to go!
Thanks!

File: contributing.md (hunk 4 above)
Comment (jatkinson1000) [reply contributing.md]: Done

File: contributing.md (hunk 5 above)
Comment (jatkinson1000) [reply contributing.md]: Done

**Review by lcimoli — APPROVED** [review]

**Pull request merged**